superset-frontend/packages/superset-ui-core/src/query/types/Column.ts

```diff
@@ -38,7 +38,7 @@ export type PhysicalColumn = string;
  * Column information defined in datasource.
  */
 export interface Column {
-  id: number;
+  id?: number;
```

Comment:
when would a query column not have an id?

Reply:
When it's of type datasource is `Query` we don't have an id for this object, since we are just saving it has a JSON blob inside `extra_json`

Reply:
got it, thanks